subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/JvmClassHasher.java

```diff
@@ -64,7 +67,8 @@ public HashCode hashClassFile(FileDetails fileDetails) {
             }
             signature = hasher.hash();
         } catch (Exception e) {
-            throw new UncheckedIOException("Could not calculate the signature for class file " + file, e);
+            signature = MALFORMED_CLASS;
```

Comment:
Fair point. But if it's our ASM analysis that is broken (and we never know), then the error message is invalid altogether.